repos/system_upgrade/el7toel8/actors/restrictedpcisscanner/libraries/restrictedpcisscanner.py

```diff
@@ -0,0 +1,161 @@
+import json
+import os
+
+import urllib3
+
+from leapp.libraries.common.config import get_env
+from leapp.libraries.stdlib import api
+from leapp.models import RestrictedPCIDevices
+
+try:
+    # python3
+    from json import JSONDecodeError  # pylint: disable=ungrouped-imports
+except ImportError:
+    # python2
+    JSONDecodeError = ValueError
+    FileNotFoundError = IOError  # pylint: disable=redefined-builtin
+
+REQUEST_TIMEOUT = 0.5  # TODO Consider moving it to the leapp env var
+CUR_DIR = os.path.dirname(os.path.abspath(__file__))
+UNSUPPORTED_DRIVER_NAMES_GLOBAL_FILE = os.path.join(
+    "/etc/leapp/files", "unsupported_driver_names.json"
+)
+UNSUPPORTED_PCI_IDS_GLOBAL_FILE = os.path.join(
+    "/etc/leapp/files", "unsupported_pci_ids.json"
+)
+UNSUPPORTED_DRIVER_NAMES_FALLBACK_FILE = os.path.join(
+    CUR_DIR, "..", "data", "unsupported_driver_names.json"
+)
+UNSUPPORTED_PCI_IDS_FALLBACK_FILE = os.path.join(
+    CUR_DIR, "..", "data", "unsupported_pci_ids.json"
+)
+
+# TODO Once the microservice for providing the data about removed hardware
+#   will be provisioned to the Red Hat costumers, then this variable default
+#   value should be overwritten
+PCIS_HOST_DEFAULT = ""
+API_VERSION = "v1"
+
+
+def get_restricted_pcis(http, update_fallback_data=False):
+    """
+    Get data about restricted driver names and unsupported pci_ids online.
+
+    The function makes API call to the web service, which provides
+    these data.
+
+    :param urllib3.PoolManager http: pool manager instance
+    :param bool update_fallback_data: if True writes the received data to
+        the corresponding json data files (inside ../data dir)
+
+    :returns: Tuple[dict, dict]
+    """
+    # TODO this constant should be above the scope of this func.
+    #   However due to the way how actor tested with CurrentActorMocked
+    #   we can't do this.
+    API_URL = "http://{host_port}/api/{api_version}/".format(
+        host_port=get_env(
+            "LEAPP_DEVEL_PCIS_HOST",
+            default=PCIS_HOST_DEFAULT,
+        ),
+        api_version=API_VERSION,
+    )
+    unsupported_driver_names = http.request(
+        "GET",
+        API_URL + "unsupported_driver_names/",
+        timeout=REQUEST_TIMEOUT,
+    )
+    unsupported_pci_ids = http.request(
+        "GET",
+        API_URL + "unsupported_pci_ids/",
+        timeout=REQUEST_TIMEOUT,
+    )
+
+    unsupported_driver_names = json.loads(
+        unsupported_driver_names.data.decode("utf-8")
+    )
+    unsupported_pci_ids = json.loads(unsupported_pci_ids.data.decode("utf-8"))
+    if update_fallback_data:
+        with open(UNSUPPORTED_DRIVER_NAMES_FALLBACK_FILE, mode="w") as f:
```

Comment:
@ZhukovGreen @fernflower  @vinzenz  aaand one thing. The write action should not be done on this path. We are not supposed to create mess under directories which are expected to contain basically just the content installed via RPMs. Open these files strictly just for reading, when no write is executed. If you need to write anything in this file, obviously it cannot be located under this directory, nor it cannot be part of the repository as it is now if I see right. As it means, that you change the installed content, which is prohibited. If you need the write operation, you can work under /tmp when it's temporary file (this case?) or you can store it under the /var/lib/leapp

Same below.

Reply:
Just thinking about that, I really do not understand the need of existence of two sources... why not to do read just from the /etc/leapp/files always? Having this, is kind of 2 sources of truth. Why that?

Reply:
@pirat89 the actor is just for reading. I just left a kind of backdoor for a test, which auto-updates the fallback source data (this test is skipped and it is documented how to use it https://github.com/oamg/leapp-repository/pull/602/files#diff-5729cbaa3149c6307578a2df2d1133ab0d2649370965d6ebc88ac45f768965c2R16-R22).

The logic behind of three steps in getting the data source is described here:
https://github.com/oamg/leapp-repository/pull/602/files#diff-5729cbaa3149c6307578a2df2d1133ab0d2649370965d6ebc88ac45f768965c2R11-R15